docs/syntax/Cadence.syntax

```diff
@@ -0,0 +1,1482 @@
+
+%scanner
+
+*
+<spaceToken>:
+        whitespace
+    ;
+
+whitespace:
+        1*whitespaceItem
+    ;
+
+whitespaceItem:
+        singleLineComment
+    |   multipleLineComment
+    |   whitespaceCharacter
+    |   lineBreak
+    ;
+
+whitespaceCharacter:
+        U+0000,U+0009,U+000B-U+000C,U+0020
+    ;
+
+lineBreak:
+        "\r" "\n"
+    |   "\r"
+    |   "\n" ;
+
+singleLineComment:
+        "/" "/" *1singleLineCommentText lineBreak
+    ;
+
+singleLineCommentText:
+        1*singleLineCommentTextItem
+    ;
+
+singleLineCommentTextItem:
+        ~"\n\r" ;
+
+multipleLineComment:
+        multipleLineCommentStart *1multipleLineCommentText multipleLineCommentEnd
+    ;
+
+multipleLineCommentStart:
+        "/" "*" ;
+
+multipleLineCommentEnd:
+        "*" "/" ;
+
+multipleLineCommentText:
+        1*multipleLineCommentTextItem
+    ;
+
+multipleLineCommentTextItem:
+        multipleLineCommentTextCharacter
+    |   multipleLineComment
+    ;
+
+multipleLineCommentTextCharacter:
+        ~multipleLineCommentStartOrEnd
+    ;
+
+multipleLineCommentStartOrEnd:
+        multipleLineCommentStart
+    |   multipleLineCommentEnd
+    ;
+
+<identifierToken>:
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz" *identifierFollower
+    ;
+
+identifierFollower:
+        "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz" ;
+
+<fixedPointToken>:
+        "0123456789" *"0123456789_" "." "0123456789" *"0123456789_"
+    ;
+
+<decimalToken>:
+        "0"
+    |   "123456789" *"0123456789_" ;
+
+<binaryToken>:
+        "0" "b" 1*"01_" ;
+
+<octalToken>:
+        "0" "o" 1*"01234567_" ;
+
+<hexadecimalToken>:
+        "0" "x" 1*"0123456789ABCDEFabcdef_" ;
+
+<invalidNumberToken>:
+        "0" "acdefghijklmnpqrstuvwyzABCDEFGHIJKLMNOPQRSTUVWXYZ" "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_"
+    ;
+
+<stringToken>:
+        "\"" *quotedTextItem "\"" ;
+
+quotedTextItem:
+        quotedTextItemCharacter
+    |   escapedCharacter
+    ;
+
+quotedTextItemCharacter:
+        ~nonquotedTextItemCharacter
+    ;
+
+nonquotedTextItemCharacter:
+        "\"\r\n\\" ;
+
+escapedCharacter:
+        escapedUnicodeScalar
+    |   escapedCCharacter
+    ;
+
+escapedUnicodeScalar:
+        "\\" "u" "{" 1*8unicodeScalarDigit "}" ;
+
+unicodeScalarDigit:
+        "0123456789abcdefABCDEF" ;
+
+escapedCCharacter:
+        "\\" "0\\tnr\"'" ;
+
+%parser
+
+*
+program:
+         *declaration( declarationSeparator ) $
```

Comment:
Could we make it more explicit that the notation here means "separated by"? The parentheses look like grouping (like in other notations), and at first glance it looks like they are unnecessary and can be removed.

Reply:
This language does not have a grouping operator.

Reply:
In that case, could you please leave a comment above, with an equivalent rule in somehting like EBNF notation? Same for the comments above (adding a comment for ranges)